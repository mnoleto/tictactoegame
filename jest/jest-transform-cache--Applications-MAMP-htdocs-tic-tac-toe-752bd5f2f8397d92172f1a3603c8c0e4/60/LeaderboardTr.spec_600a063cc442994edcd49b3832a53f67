/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_LeaderboardTr = require('./LeaderboardTr'); /* istanbul ignore next */var _LeaderboardTr2 = _interopRequireDefault(_LeaderboardTr);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// LeaderboardTr is responsible for render the game grid
describe('LeaderboardTr', function () {
  var props = /* istanbul ignore next */void 0,mountedLeaderboardTr = /* istanbul ignore next */void 0;
  var leaderboardTr = function leaderboardTr() {
    if (!mountedLeaderboardTr) {
      mountedLeaderboardTr = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_LeaderboardTr2.default, props));

    }
    return mountedLeaderboardTr;
  };

  beforeEach(function () {
    props = {
      defeats: 0,
      draws: 0,
      name: 'Player 1',
      position: 0,
      wins: 0 };

    mountedLeaderboardTr = undefined;
  });

  it('should render the LeaderboardTr', function () {
    var leaderboardTrWrapper = /* istanbul ignore next */(0, _enzyme.shallow)( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_LeaderboardTr2.default, props));
    expect(leaderboardTrWrapper).toMatchSnapshot();
  });

  // it('tr contains everything else that gets rendered', () => {
  //   let leaderboardTrElement = leaderboardTr().find('tr');
  //   expect(leaderboardTrElement.children()).toEqual(leaderboardTr().children());
  // });

  it('should have 5 td', function () {
    var tdElements = leaderboardTr().find('tr');
    //   expect(tdElements.length).toEqual(5);
  });

  it('should the first td has the player name', function () {

  });

  it('should the second td has the wins score', function () {

  });

  it('should the third td has the defeats score', function () {

  });

  it('should the third td has the draws score', function () {

  });

  it('should order the players by score', function () {

  });
});