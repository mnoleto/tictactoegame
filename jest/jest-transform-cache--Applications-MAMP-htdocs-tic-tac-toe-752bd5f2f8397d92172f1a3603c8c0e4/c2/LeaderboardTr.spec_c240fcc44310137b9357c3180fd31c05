/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_enzyme = require('enzyme');
var /* istanbul ignore next */_LeaderboardTr = require('./LeaderboardTr'); /* istanbul ignore next */var _LeaderboardTr2 = _interopRequireDefault(_LeaderboardTr);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

// LeaderboardTr is responsible for render the game grid
describe('LeaderboardTr', function () {
  var props = /* istanbul ignore next */void 0,mountedLeaderboardTr = /* istanbul ignore next */void 0;
  var leaderboardTr = function leaderboardTr() {
    if (!mountedLeaderboardTr) {
      mountedLeaderboardTr = /* istanbul ignore next */(0, _enzyme.mount)(
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'table', /* istanbul ignore next */null,
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'tbody', /* istanbul ignore next */null,
          /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_LeaderboardTr2.default, props))));



    }
    return mountedLeaderboardTr;
  };

  beforeEach(function () {
    props = {
      defeats: 0,
      draws: 0,
      name: 'Player 1',
      position: 0,
      wins: 0 };

    mountedLeaderboardTr = undefined;
  });

  it('should render the LeaderboardTr', function () {
    var leaderboardTrWrapper = /* istanbul ignore next */(0, _enzyme.shallow)(
    /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'table', /* istanbul ignore next */null,
      /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'tbody', /* istanbul ignore next */null,
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_LeaderboardTr2.default, props))));



    expect(leaderboardTrWrapper).toMatchSnapshot();
  });

  it('should have 5 td', function () {
    var tdElements = leaderboardTr().find('td');
    expect(tdElements.length).toEqual(5);
  });

  it('should the first td has the player position', function () {
    var trRenderer = leaderboardTr();
    var trElement = trRenderer.find('tr');
    var firstTd = trElement.childAt(0);
    expect(firstTd.hasClass('position')).toBeTruthy();
  });

  it('should the second td has the player name', function () {
    var trRenderer = leaderboardTr();
    var trElement = trRenderer.find('tr');
    var firstTd = trElement.childAt(1);
    expect(firstTd.hasClass('name')).toBeTruthy();
  });

  it('should the third td has the wins score', function () {
    var trRenderer = leaderboardTr();
    var trElement = trRenderer.find('tr');
    var firstTd = trElement.childAt(2);
    expect(firstTd.hasClass('wins')).toBeTruthy();
  });

  it('should the forth td has the defeats score', function () {
    var trRenderer = leaderboardTr();
    var trElement = trRenderer.find('tr');
    var firstTd = trElement.childAt(3);
    expect(firstTd.hasClass('defeats')).toBeTruthy();
  });

  it('should the fifth td has the draws score', function () {
    var trRenderer = leaderboardTr();
    var trElement = trRenderer.find('tr');
    var firstTd = trElement.childAt(4);
    expect(firstTd.hasClass('draws')).toBeTruthy();
  });
});