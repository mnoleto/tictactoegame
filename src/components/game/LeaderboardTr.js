import React, { Component, PropTypes } from 'react';

// LeaderboardTr is responsible for render the leaderboard table line
const LeaderboardTr = ({ defeats, draws, name, position, wins }) => {
  return (
    <tr>
      <td className="position">{position}</td>
      <td className="name">{name}</td>
      <td className="wins">{wins}</td>
      <td className="defeats">{defeats}</td>
      <td className="draws">{draws}</td>
    </tr>
  );
}

LeaderboardTr.propTypes = {
  defeats: PropTypes.number.isRequired,
  draws: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  position: PropTypes.number.isRequired,
  wins: PropTypes.number.isRequired
};

export default LeaderboardTr;