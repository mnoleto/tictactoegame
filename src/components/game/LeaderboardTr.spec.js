import React from 'react';
import { mount, shallow } from 'enzyme';
import LeaderboardTr from './LeaderboardTr';

// LeaderboardTr is responsible for render the leaderboard table line
describe('<LeaderboardTr />', () => {
  let props, mountedLeaderboardTr;
  const leaderboardTr = () => {
    if(!mountedLeaderboardTr) {
      mountedLeaderboardTr = mount(
        <table>
          <tbody>
            <LeaderboardTr {...props} />
          </tbody>
        </table>
      );
    }
    return mountedLeaderboardTr;
  };

  beforeEach(() => {
    props = {
      defeats: 0,
      draws: 0,
      name: 'Player 1',
      position: 0,
      wins: 0
    };
    mountedLeaderboardTr = undefined;
  });

  it('should render the LeaderboardTr', () => {
    const leaderboardTrWrapper = shallow(
      <table>
        <tbody>
          <LeaderboardTr {...props} />
        </tbody>
      </table>
    );
    expect(leaderboardTrWrapper).toMatchSnapshot();
  });

  it('should have 5 td', () => {
    const tdElements = leaderboardTr().find('td');
    expect(tdElements.length).toEqual(5);
  });

  it('should the first td has the player position', () => {
    const trRenderer = leaderboardTr();
    const trElement = trRenderer.find('tr');
    const firstTd = trElement.childAt(0);
    expect(firstTd.hasClass('position')).toBeTruthy();
  });

  it('should the second td has the player name', () => {
    const trRenderer = leaderboardTr();
    const trElement = trRenderer.find('tr');
    const firstTd = trElement.childAt(1);
    expect(firstTd.hasClass('name')).toBeTruthy();
  });

  it('should the third td has the wins score', () => {
    const trRenderer = leaderboardTr();
    const trElement = trRenderer.find('tr');
    const firstTd = trElement.childAt(2);
    expect(firstTd.hasClass('wins')).toBeTruthy();
  });

  it('should the forth td has the defeats score', () => {
    const trRenderer = leaderboardTr();
    const trElement = trRenderer.find('tr');
    const firstTd = trElement.childAt(3);
    expect(firstTd.hasClass('defeats')).toBeTruthy();
  });

  it('should the fifth td has the draws score', () => {
    const trRenderer = leaderboardTr();
    const trElement = trRenderer.find('tr');
    const firstTd = trElement.childAt(4);
    expect(firstTd.hasClass('draws')).toBeTruthy();
  });
});