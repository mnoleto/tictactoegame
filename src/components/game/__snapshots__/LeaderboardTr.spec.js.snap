exports[`<LeaderboardTr /> should render the LeaderboardTr 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <table>
    <tbody>
        <LeaderboardTr
            defeats={0}
            draws={0}
            name="Player 1"
            position={0}
            wins={0} />
    </tbody>
</table>,
  "nodes": Array [
    <table>
      <tbody>
            <LeaderboardTr
                  defeats={0}
                  draws={0}
                  name="Player 1"
                  position={0}
                  wins={0} />
      </tbody>
</table>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": null,
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <table>
    <tbody>
        <LeaderboardTr
            defeats={0}
            draws={0}
            name="Player 1"
            position={0}
            wins={0} />
    </tbody>
</table>,
}
`;
